packages/react-components/lib/robots/robot-table.tsx

```diff
@@ -60,44 +60,11 @@ const StyledPaper = styled((props: PaperProps) => <Paper {...props} />)(({ theme
 
 interface RobotRowProps {
   robot: RobotState;
+  fetchSelectedTask?: (taskId: string) => Promise<TaskState | undefined>;
```

Comment:
I would prefer if `react-components` don't require fetching of data. This can lead of some usability problems, for example, imagine if the robot detail panel also requires fetching of data, then the robot page would fetch the same data multiple times, with possibly different results each time.

Instead, it would be better to wrap the `RobotState` and its current `TaskState`, hypothetically called `RobotData` (any better name which doesn't conflict with rmf model names?)

```tsx
interface RobotData {
  state: RobotState;
  current_task_state?: TaskState;
}
```

Reply:
there was a `VerboseRobot` that might no longer be used as is now, perhaps we can reuse that?

Reply:
i just realized the fetching of data already exists in robot-table and robot-panel: the fetchSelectedTask is defined in robot-page and passed into all the robot related react-components. Should i move fetchSelectedTask out of all the react-components, in that case?

Reply:
> i just realized the fetching of data already exists in robot-table and robot-panel: the fetchSelectedTask is defined in robot-page and passed into all the robot related react-components. Should i move fetchSelectedTask out of all the react-components, in that case?

Fetch data will be needed at some point, but better to put it in dashboard. Components here should be re-usable and having them handle fetching of data puts an assumption on how the app manages data.

> there was a VerboseRobot that might no longer be used as is now, perhaps we can reuse that?

Naming things is always hard, to me, `VerboseRobot` doesn't exactly explain what it represents. On another note, what if suppose that the robot row needs `RobotState` and `TaskState`, and the robot detail also needs `TaskLog`, how should be name the interfaces?

One way to avoid this naming conflict is to not name them in the first place, but that will hurt readability, e.g.

```ts
export interface RobotRowProps {
  data: { state: RobotState, current_task: TaskState }
  ...
}

export interface RobotDetailProps {
  data: { state: RobotState, current_task_state: TaskState, current_task_log: TaskLog }
  ...
}
```

Another way is to drop the reliance on rmf models, e.g.

```ts
export interface RobotRowProps {
  name: string,
  active_task_duration: number,
  battery: number,
  ...
}
```

Theoricatically, this would be the most robust approach, unused properties will not cause needless re-renders (with the current approach, e.g. if the robot location changes, it will cause a needless re-render even though the result is the same since we are not showing the location) and things will not break when rmf models change (it will still break in dashboard, but at least react-components will not be affected).

Reply:
i went ahead and used "verboserobot" but it should be straightforward to refactor this name